### ISSUE TITLE
iOS Library Suggestion

### ISSUE THREAD

**Issue by sankar2389** [baronha/react-native-photo-editor#3]
Hi, 
Instead of Brightroom ios library you can use the below mentioned library.
https://github.com/longitachi/ZLImageEditor
This library has a matching modules respective to the Android Native library.

**Comment by baronha** [comment]
> Hi,
> Instead of Brightroom ios library you can use the below mentioned library.
> https://github.com/longitachi/ZLImageEditor
> This library has a matching modules respective to the Android Native library.
I used to use it. although it is extremely suitable for android but it uses high memory. Maybe I will use (GPUImage)[https://github.com/BradLarson/GPUImage3] for a completely new rewrite for iOS.

**Comment by jphilipstevens** [comment]
I think for ios this one would be a big upgrade from BrightRoom

**Comment by baronha** [comment]
> I think for ios this one would be a big upgrade from BrightRoom

Do you think I should replace BrightRoom with ZLImageEditor?

**Comment by sankar2389** [comment]
Actually I already wrote the binder for ReactNative with ZLImageEditor. If you want I can share my code 👍🏼

**Comment by jphilipstevens** [comment]
@sankar2389 that would be great actually! I would like to  POC this lib with ZLImageEditor

**Comment by sankar2389** [comment]
https://drive.google.com/file/d/1_3Tyi9FFZYKE3M5rSiXcjl6A3Oi2peQS/view?usp=drivesdk

**Comment by baronha** [comment]
> https://drive.google.com/file/d/1_QMPEo9IWPDIN9yWjuifQ2bMxlHY8SlF/view?usp=drivesdk

OMG. I have seen the file you sent. But it's better to fork it to a separate branch and request a pull request. I will rewrite Docs for this lib. You will be one of the contributors

**Comment by sankar2389** [comment]
sure

**Comment by baronha** [comment]
Hi guy. I used ZLImageEditor in version 1.0.0 but kept Filter by LUT Color 🌈

**Comment by filipef101** [comment]
@baronha Any chance for contrast, brightness adjustments? It seems ZLImageEditor doesn't support it right?

**Comment by baronha** [comment]
> @baronha Any chance for contrast, brightness adjustments? It seems ZLImageEditor doesn't support it right?

Yup. They can be done, but there are more things to prioritize than them.

**Comment by filipef101** [comment]
@baronha I don't see anything on ZLImageEditor that allows for contrast, brighness, sharpness etc editing, did I miss something?

**Comment by baronha** [comment]
> @baronha I don't see anything on ZLImageEditor that allows for contrast, brighness, sharpness etc editing, did I miss something?

@filipef101 We'll do it as a filter with a slider like how Brightroom implemented it. Filter filter using LUT color I also used this way. You can refer to [here](https://github.com/muukii/Brightroom/tree/main/Sources/BrightroomEngine/Filter). If you have time. You try to implement it into this library.

**Comment by filipef101** [comment]
Will just use brightroom since I don't need the stickers, text and drawing
Btw, on android I noticed this ticket: https://github.com/burhanrashid52/PhotoEditor/issues/10 

Maybe somehow it might be better to use one of these:

https://github.com/AlirezaGhanbarinia/EditPhoto

https://github.com/ravi8x/AndroidPhotoFilters -> kind of similar to light room
(https://www.androidhive.info/2017/11/android-building-image-filters-like-instagram/)

https://github.com/cats-oss/android-gpuimage

Or some other that i've missed

Toughts? Do you think any of those can work?

**Comment by baronha** [comment]
> Will just use brightroom since I don't need the stickers, text and drawing Btw, on android I noticed this ticket: [burhanrashid52/PhotoEditor#10](https://github.com/burhanrashid52/PhotoEditor/issues/10)
> 
> Maybe somehow it might be better to use one of these:
> 
> https://github.com/AlirezaGhanbarinia/EditPhoto
> 
> https://github.com/ravi8x/AndroidPhotoFilters -> kind of similar to light room (https://www.androidhive.info/2017/11/android-building-image-filters-like-instagram/)
> 
> https://github.com/cats-oss/android-gpuimage
> 
> Or some other that i've missed
> 
> Toughts? Do you think any of those can work?

Hi @filipef101. Perhaps you are in need of things related to color filters. It's great that you can build a library of it. You can use [GPUImage3(swift)](https://github.com/BradLarson/GPUImage3) or [GPUImage(Objc)](https://github.com/BradLarson/GPUImage) in iOS and [android -gpuimage](https://github.com/cats-oss/android-gpuimage) in Android. Or simply you just need Brighroom and [AndroidPhotoFilters](https://github.com/ravi8x/AndroidPhotoFilters).

**Comment by filipef101** [comment]
Might turn that into a library based on your androidPhotoFilters!

With that message I wanted to shed light that the Android library might not be great, as it will result in low resolution images, since basically it is taking a picture of the screen

**Comment by filipef101** [comment]
Just another blurt of  information @baronha, saw this https://github.com/Zomato/AndroidPhotoFilters tought could be useful for your implementation of LUTs on android

**Comment by baronha** [comment]
> Just another blurt of information @baronha, saw this https://github.com/Zomato/AndroidPhotoFilters tought could be useful for your implementation of LUTs on android

Thank you @filipef101. I'm deciding between [Android-GPUImage](https://github.com/cats-oss/android-gpuimage) and [easyLUT](https://github.com/dntks/easyLUT). But easyLUT hasn't been maintained for a long time. And GPUImage is too big

**Comment by prashant334434** [comment]
Error opening photo editor: [Error: Load image failed: /Users/prashant/Library/Developer/CoreSimulator/Devices/38EE6BAE-AD54-439A-AE30-11525F6896B3/data/Containers/Data/Application/32B8C5B7-8091-4FF3-BE61-4FA7CF94E8F6/tmp/react-native-image-crop-picker/73AFC5C5-6315-4B3F-A6C5-D533C588FFD7.jpg]. 


i am getting this error when i follow all the steps but still not opening the editor
